src/generators/default_map_generator_job.cpp

```diff
@@ -273,9 +462,8 @@ bool default_map_generator_job::generate_river_internal(const height_map& height
 		LOG_NG << "generating river...\n";
 
 		// Generate the river
-		for(std::vector<location>::const_iterator i = river.begin();
-			i != river.end(); ++i) {
-			terrain[i->x][i->y] = t_translation::SHALLOW_WATER;
+		for(auto i : river) {
```

Comment:
Auto is nice, but sometimes it's nicer to make it explicitly clear what type you're working with. Just keep that in mind, no need to change anything (unless you think something is confusing with auto).